### ISSUE TITLE
Declarative Agent Declaration Skips Default Checkout

### ISSUE THREAD

**Issue by giles-jason** [jenkinsci/amazon-ecs-plugin#94]
It appears that when defining an agent using the declarative syntax, the default checkout functionality that comes with declarative pipelines is disabled.

I suspect the issue might be in `com.cloudbees.jenkins.plugins.amazonecs.pipeline.ECSDeclarativeAgentScript` but can't be certain. That class appears to import `org.jenkinsci.plugins.pipeline.modeldefinition.agent.CheckoutScript` but does not do anything with it.

There is a simple enough workaround (explicitly disable default checkout and add a checkout stage):

```
pipeline {
  agent {
    ecs {
      inheritFrom 'my-base-agent-label'
      label 'my-custom-agent-label'
    }
  }

  options {
    skipDefaultCheckout()
  }

  stages {
    stage ('Checkout') {
      steps {
        checkout scm
      }
    }

    ...more stages for the pipeline...

  }
}
```

Obviously it would be preferred though if the default checkout functionality was preserved when using a declarative definition for an ECS agent.

**Comment by pgarbe** [comment]
Thanks for reporting. 

I guess that `CheckoutScript.doCheckout(script, describable, describable.customWorkspace)` is missing inside the `script.node()` [block](https://github.com/jenkinsci/amazon-ecs-plugin/blob/5ea651d2e0c6cb827b540fe46d8ca713381b57fd/src/main/resources/com/cloudbees/jenkins/plugins/amazonecs/pipeline/ECSDeclarativeAgentScript.groovy#L51)

**Comment by webratz** [comment]
There should be a beta of 1.21 in the next few minutes maybe you can check if its still there. i remember there were some changes

**Comment by mdouglass** [comment]
Looks like 1.21 beta did not fix it, I have sent PR #115 to address it
